dmarc/src/main/java/org/apache/james/dmarc/DMARCVerifier.java

```diff
@@ -0,0 +1,137 @@
+ * regarding copyright ownership.  The ASF licenses this file                 *
+ * to you under the Apache License, Version 2.0 (the                          *
+ * "License"); you may not use this file except in compliance                 *
+ * with the License.  You may obtain a copy of the License at                 *
+ *                                                                            *
+ *   http://www.apache.org/licenses/LICENSE-2.0                               *
+ *                                                                            *
+ * Unless required by applicable law or agreed to in writing,                 *
+ * software distributed under the License is distributed on an                *
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY                     *
+ * KIND, either express or implied.  See the License for the                  *
+ * specific language governing permissions and limitations                    *
+ * under the License.                                                         *
+ ******************************************************************************/
+package org.apache.james.dmarc;
+
+import org.apache.james.dmarc.exceptions.DmarcException;
+import org.apache.james.mime4j.dom.Message;
+
+public class DMARCVerifier {
+    public static final String FROM = "From";
+    private final String _dmarcResponse;
+    private final String _dmarcNonResponse;
+    private final PublicKeyRecordRetrieverDmarc _recordRetriever;
+
+    public DMARCVerifier(String dmarcResponse, String dmarcNonResponse, PublicKeyRecordRetrieverDmarc recordRetriever) {
+        _dmarcResponse = dmarcResponse;
+        _dmarcNonResponse = dmarcNonResponse;
+        _recordRetriever = recordRetriever;
+    }
+
+    public String runDmarcCheck(Message message, String spfHeaderText, String
+            spfDomain, String dkimResult, String dkimDomain){
+             // Combine SPF + DKIM results with From: domain
+        // 1. Extract RFC5322.From domain from the From header of the message
+        String shortSpfResult = spfHeaderText.split(" ")[0];
+        String fromHeader = message.getHeader().getField(FROM).getBody();
+        String fromDomain = extractDomain(fromHeader);
+        if (fromDomain == null || fromDomain.isEmpty()) {
+            return _dmarcNonResponse + "unknown";
+        }
+
+        // 2. Fetch DMARC record from DNS
+        String dmarcRecord = _recordRetriever.getDmarcRecord(fromDomain);
+        if (dmarcRecord == null) {
+            return _dmarcNonResponse + fromDomain;
+        }
+
+        // Parse DMARC policy
+        String policy = parseTag(dmarcRecord, "p"); // p=none|quarantine|reject
+        String aspf = parseTag(dmarcRecord, "aspf"); // "s" or "r" for strict or relaxed domain alignment; default is "r"
+        String adkim = parseTag(dmarcRecord, "adkim"); // "s" or "r" for strict or relaxed domain alignment; default is "r"
```

Comment:
Maybe we can be more efficient by doing one pass to build a `Map<String, String>` of tags then do maps lookup instead of repetitive parsing?